### DISCUSSION TITLE
仿Bilibili客户端和网页的一个导航，现已实现大部分功能，缺个MouseOver的动画效果，求个思路

### DISCUSSION THREAD

**Discussion by edward-chung** [BYJRK/DotNet-Discussions#58] [WPF 开发]
![6](https://user-images.githubusercontent.com/9280859/225932044-bda678c4-1d08-4b82-840b-d6a9a0fed9ba.gif)

网页端个人空间的导航MouseOver的效果是鼠标放上去滑块移动到当前位置，移开会回原位

![7](https://user-images.githubusercontent.com/9280859/225932796-4e85c544-f935-4858-9c09-8f2083e45888.gif)

`protected override void OnAttached()
    {
        AssociatedObject.Loaded += (_, _) =>
        {
            sliding = (FrameworkElement)AssociatedObject.FindName(SlidingName);
            Children = GetChildren<RadioButton>().ToList();
            foreach (var child in Children)
            {
                child.Checked += OnRadioButtonChecked;
                if ((bool)child.IsChecked)
                {
                    BeginAnimation(child);
                }
            }
        };
    }`

但是WPF Behavior不好注册这个事件。求个实现思路

github地址[https://github.com/edward-chung/WPF.Demo](https://github.com/edward-chung/WPF.Demo)

**Comment by eijjxx** [comment]
注册MouseLeave。MouseLeave的时候直接Slider的位置置为0
